django/db/models/sql/query.py

```diff
@@ -390,10 +390,19 @@ def get_aggregation(self, using, added_aggregate_names):
             from django.db.models.sql.subqueries import AggregateQuery
             outer_query = AggregateQuery(self.model)
             inner_query = self.clone()
-            if not has_limit and not self.distinct_fields:
-                inner_query.clear_ordering(True)
             inner_query.select_for_update = False
             inner_query.select_related = False
+            if not has_limit and not self.distinct_fields:
+                inner_query.clear_ordering(True)
+            if not inner_query.distinct:
+                # If the inner query uses default select, and it has some aggregate annotations,
```

Comment:
remove first comma
(also as someone not intimately familiar with the ORM, I'm not quite sure what "default select" means (I assume it's a reference to `default_cols` -- maybe no further explanation is needed).